service/history/workflowExecutionContext.go

```diff
@@ -239,6 +239,15 @@ func (c *workflowExecutionContext) updateHelper(builder *historyBuilder, transfe
 		replicationTasks = append(replicationTasks, c.msBuilder.createReplicationTask())
 	}
 
+	// this is the current failover version
```

Comment:
ShardContext.Update already has the loop to go over all tasks and assign TaskIDs.  Can we use the same loop to assign version?